ios/browser/api/features/brave_feature_list.mm

```diff
@@ -0,0 +1,334 @@
+             "BraveWalletZCash",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kConstellation,
+             "BraveP3AConstellation",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kConstellationEnclaveAttestation,
+             "BraveP3AConstellationEnclaveAttestation",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kCosmeticFilteringExtraPerfMetrics,
+             "CosmeticFilteringExtraPerfMetrics",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kCosmeticFilteringJsPerformance,
+             "CosmeticFilteringJsPerformance",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kCosmeticFilteringSyncLoad,
+             "CosmeticFilterSyncLoad",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kFederatedLearning,
+             "BraveFederated",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kGeminiFeature,
+             "BraveRewardsGemini",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kIpfsFeature,
+             "Ipfs",
+#if BUILDFLAG(ENABLE_IPFS)
+             base::FEATURE_ENABLED_BY_DEFAULT
+#else
+             base::FEATURE_DISABLED_BY_DEFAULT
+#endif
+);
+
+BASE_FEATURE(kNTP, "BraveSearchNTP", base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kNativeBraveWalletFeature,
+             "NativeBraveWallet",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kOmniboxBanner,
+             "BraveSearchOmniboxBanner",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kOmniboxButton,
+             "BraveSearchOmniboxButton",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kOtherJSONDeprecation,
+             "BraveP3AOtherJSONDeprecation",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kPageGraph, "PageGraph", base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kPlaylist, "Playlist", base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kPlaylistFakeUA,
+             "PlaylistFakeUA",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kShowRewardsCard,
+             "ShowRewardsCard",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kSkusFeature, "SkusFeature", base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kSpeedreaderFeature,
+             "Speedreader",
+#if BUILDFLAG(IS_ANDROID)
+             base::FEATURE_DISABLED_BY_DEFAULT);
+#else
+             base::FEATURE_ENABLED_BY_DEFAULT);
+#endif
+
+BASE_FEATURE(kTypicalJSONDeprecation,
+             "BraveP3ATypicalJSONDeprecation",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kUseDevUpdaterUrl,
+             "UseDevUpdaterUrl",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
+BASE_FEATURE(kVBatNoticeFeature,
+             "BraveRewardsVBatNotice",
+             base::FEATURE_ENABLED_BY_DEFAULT);
+
+BASE_FEATURE(kVerboseLoggingFeature,
+             "BraveRewardsVerboseLogging",
+             base::FEATURE_DISABLED_BY_DEFAULT);
```

Comment:
why all features here are redeclared? why not just reference features directly in `ios/browser/api/features/features.mm`?

the current impl looks wrong and will quickly get out of sync.

Reply:
> This is what chrome does on Android. These are autogenerated as well using some code I wrote. There's not really any other way to export all features for iOS.
> 
> Chrome on Android does similar by exporting a feature mapping via JNI.

it's fine to map existing features to JNI/Swift, but it's not fine to duplicate feature declaration, which `default_state`, `name` or something else can divert at any point.

does Chrome on Android redeclares `base::Feature` instances or it maps Java feature checks to the existing features declared in the native library?

my main concern here is that you are redeclaring existing `base::Features` for no obvious reason.

Reply:
> Hmm okay. Removed and fixed and rebased. Chrome Android dumps all the android features into the same namespace and declares them all: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/flags/android/chrome_feature_list.h

thanks. I see that Chromium declares only Android features there, but they just expose existing features directly via `kFeaturesExposedToJava`. That's correct and make perfect sense.